sa/sa.go

```diff
@@ -1075,6 +1075,54 @@ func addFQDNSet(tx *gorp.Transaction, names []string, serial string, issued time
 	})
 }
 
+// addOrderFQDNSet creates a new OrderFQDNSet row using the provided
+// information. This function doesn't accept a transaction because it must be
+// performed after the add transaction has completed in order for the orderID to
```

Comment:
Are you sure about this? I thought that autoincrement keys did get assigned a value even inside a transaction.

Reply:
You're right :blush: I'm not sure where I got this idea. Changed `addOrderFQDNSet` to operate on the open transaction from the order creation in 57131c2

Reply:
Assuming that autoincrement ids don't get assigned until finalization makes sense, because otherwise you'd get gaps in your autoincrement sequence if you roll back. However, MySQL (and maybe SQL in general?) made the opposite design choice, presumably because of cases like this: the internal autoincrement counter gets incremented immediately and globally, even if you're inside a transaction, and if you roll back, there's a gap. Good example of a leaky abstraction. :-)